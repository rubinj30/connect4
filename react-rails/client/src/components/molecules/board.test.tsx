import React from 'react';
import { shallow } from 'enzyme';
import { Board } from './board.component';

describe('Board component', () => {
    let fixture, result;
    beforeEach(() => {
        fixture = <Board />;
        result = shallow(fixture);
    });
    it('should render properly', () => {
        expect(result).toBeDefined();
        expect(result).toMatchSnapshot();
    });
    describe('functions', () => {
        it('changeTurn should change the currentTurn from "R" to "B" and vice versa', () => {
            result.setState({ currentTurn: 'R' });
            const instance = result.instance();
            instance.changeTurn();
            expect(result.state().currentTurn).toEqual('B');
            instance.changeTurn();
            expect(result.state().currentTurn).toEqual('R');
        });
        it('dropPieceInColumn should call find first blank space and replace it with the piece in currentTurn', () => {
            
        });
        it('replaceColumn should call find first blank space and replace it with the piece in currentTurn', () => {

        });
    });
});
