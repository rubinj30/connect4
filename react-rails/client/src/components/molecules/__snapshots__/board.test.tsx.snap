// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        Array [
          <Column
            column={
              Array [
                " ",
                " ",
                " ",
                " ",
                " ",
                " ",
              ]
            }
            currentTurn="B"
            dataIndex={0}
            handleClick={[Function]}
            win={false}
          />,
          <Column
            column={
              Array [
                " ",
                " ",
                " ",
                " ",
                " ",
                " ",
              ]
            }
            currentTurn="B"
            dataIndex={1}
            handleClick={[Function]}
            win={false}
          />,
          <Column
            column={
              Array [
                " ",
                " ",
                " ",
                " ",
                " ",
                " ",
              ]
            }
            currentTurn="B"
            dataIndex={2}
            handleClick={[Function]}
            win={false}
          />,
          <Column
            column={
              Array [
                " ",
                " ",
                " ",
                " ",
                " ",
                " ",
              ]
            }
            currentTurn="B"
            dataIndex={3}
            handleClick={[Function]}
            win={false}
          />,
          <Column
            column={
              Array [
                " ",
                " ",
                " ",
                " ",
                " ",
                " ",
              ]
            }
            currentTurn="B"
            dataIndex={4}
            handleClick={[Function]}
            win={false}
          />,
          <Column
            column={
              Array [
                " ",
                " ",
                " ",
                " ",
                " ",
                " ",
              ]
            }
            currentTurn="B"
            dataIndex={5}
            handleClick={[Function]}
            win={false}
          />,
          <Column
            column={
              Array [
                " ",
                " ",
                " ",
                " ",
                " ",
                " ",
              ]
            }
            currentTurn="B"
            dataIndex={6}
            handleClick={[Function]}
            win={false}
          />,
        ],
      ],
      "className": "bg-blue h-100 pa3 flex flex-column items-center justify-center board",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "column": Array [
            " ",
            " ",
            " ",
            " ",
            " ",
            " ",
          ],
          "currentTurn": "B",
          "dataIndex": 0,
          "handleClick": [Function],
          "win": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "column": Array [
            " ",
            " ",
            " ",
            " ",
            " ",
            " ",
          ],
          "currentTurn": "B",
          "dataIndex": 1,
          "handleClick": [Function],
          "win": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "column": Array [
            " ",
            " ",
            " ",
            " ",
            " ",
            " ",
          ],
          "currentTurn": "B",
          "dataIndex": 2,
          "handleClick": [Function],
          "win": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "column": Array [
            " ",
            " ",
            " ",
            " ",
            " ",
            " ",
          ],
          "currentTurn": "B",
          "dataIndex": 3,
          "handleClick": [Function],
          "win": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "column": Array [
            " ",
            " ",
            " ",
            " ",
            " ",
            " ",
          ],
          "currentTurn": "B",
          "dataIndex": 4,
          "handleClick": [Function],
          "win": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "column": Array [
            " ",
            " ",
            " ",
            " ",
            " ",
            " ",
          ],
          "currentTurn": "B",
          "dataIndex": 5,
          "handleClick": [Function],
          "win": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "column": Array [
            " ",
            " ",
            " ",
            " ",
            " ",
            " ",
          ],
          "currentTurn": "B",
          "dataIndex": 6,
          "handleClick": [Function],
          "win": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [
            <Column
              column={
                Array [
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
              currentTurn="B"
              dataIndex={0}
              handleClick={[Function]}
              win={false}
            />,
            <Column
              column={
                Array [
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
              currentTurn="B"
              dataIndex={1}
              handleClick={[Function]}
              win={false}
            />,
            <Column
              column={
                Array [
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
              currentTurn="B"
              dataIndex={2}
              handleClick={[Function]}
              win={false}
            />,
            <Column
              column={
                Array [
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
              currentTurn="B"
              dataIndex={3}
              handleClick={[Function]}
              win={false}
            />,
            <Column
              column={
                Array [
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
              currentTurn="B"
              dataIndex={4}
              handleClick={[Function]}
              win={false}
            />,
            <Column
              column={
                Array [
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
              currentTurn="B"
              dataIndex={5}
              handleClick={[Function]}
              win={false}
            />,
            <Column
              column={
                Array [
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                  " ",
                ]
              }
              currentTurn="B"
              dataIndex={6}
              handleClick={[Function]}
              win={false}
            />,
          ],
        ],
        "className": "bg-blue h-100 pa3 flex flex-column items-center justify-center board",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "column": Array [
              " ",
              " ",
              " ",
              " ",
              " ",
              " ",
            ],
            "currentTurn": "B",
            "dataIndex": 0,
            "handleClick": [Function],
            "win": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "column": Array [
              " ",
              " ",
              " ",
              " ",
              " ",
              " ",
            ],
            "currentTurn": "B",
            "dataIndex": 1,
            "handleClick": [Function],
            "win": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "column": Array [
              " ",
              " ",
              " ",
              " ",
              " ",
              " ",
            ],
            "currentTurn": "B",
            "dataIndex": 2,
            "handleClick": [Function],
            "win": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "column": Array [
              " ",
              " ",
              " ",
              " ",
              " ",
              " ",
            ],
            "currentTurn": "B",
            "dataIndex": 3,
            "handleClick": [Function],
            "win": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "column": Array [
              " ",
              " ",
              " ",
              " ",
              " ",
              " ",
            ],
            "currentTurn": "B",
            "dataIndex": 4,
            "handleClick": [Function],
            "win": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "column": Array [
              " ",
              " ",
              " ",
              " ",
              " ",
              " ",
            ],
            "currentTurn": "B",
            "dataIndex": 5,
            "handleClick": [Function],
            "win": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "column": Array [
              " ",
              " ",
              " ",
              " ",
              " ",
              " ",
            ],
            "currentTurn": "B",
            "dataIndex": 6,
            "handleClick": [Function],
            "win": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
